### PR TITLE
feat(checkbox): indeterminate state

### PR DESCRIPTION
**Pull request by dantasgmg**
**Describe pull-request**  
Adding indeterminate state to Tegel checkbox component

**How to test**  
1. Run storybook
2. Go to checkbox component
3. Go to docs
4.  Set the indeterminate toggle to true/false
5.  Repeat step 4 for dark and light mode

### CODE REVIEW

File: core/src/components/checkbox/sdds-checkbox.stories.tsx [hunk 1]
```diff
@@ -48,19 +48,31 @@ export default {
...
+    indeterminate: {
+      name: 'Indeterminate',
+      description: 'Set the checkbox state to indeterminate',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        defaultValue: { summary: false },
+      },
+    },
```
Comment (mJarsater) [inline core/src/components/checkbox/sdds-checkbox.stories.tsx:60]: Really great stuff! I would recommend adding a "if" here to not let the user set the checkbox both checked and indeterminate.

Something like:
if: { arg: 'checked', neq: true },

File: core/src/components/checkbox/sdds-checkbox.scss [hunk 2]
```diff
@@ -127,5 +127,40 @@
...
+      &:disabled,
+      &.disabled {
+        &:hover::before,
+        &::before {
+          display: block;
+          width: 16px;
+          height: 16px;
+          left: 4px;
+          top: 4px;
+          border: 1px solid #b0b7c4;
```
Comment (timrombergjakobsson) [inline core/src/components/checkbox/sdds-checkbox.scss:155]:
```suggestion
          border: 1px solid var(--tds-grey-500);
```

**Comment by theJohnnyMe** [comment]
I did rebase the main branch and forced pushed changes.

File: core/src/components/checkbox/sdds-checkbox.scss (hunk 2 above)
Comment (mJarsater) [inline core/src/components/checkbox/sdds-checkbox.scss:155]: I think a lot of these styles should be for both checked/indeterminate state. Could reduce the amount of css

Comment (mJarsater) [reply core/src/components/checkbox/sdds-checkbox.scss:155]: Might even need to be a scss variable tied to the checkbox?

**Comment by theJohnnyMe** [comment]
Closing it for now.

**Pull request closed without merging**